README.md

```diff
@@ -198,6 +199,15 @@ A `TraceApi` object is guaranteed to be returned by both of these calls, even if
 
 A fully detailed overview of the `TraceApi` object is available [here](doc/trace-api.md).
 
+## How does automatic tracing work?
+
```

Comment:
Fairly strongly. As this section is meant to explain how automatic tracing works, I think it would read better to talk about injecting code right away, and then go into the details. Right now it feels like we're jumping straight into the details, without a high-level overview first.